core/templates/pages/profile-page/profile-page.component.ts

```diff
@@ -110,23 +110,28 @@ angular.module('oppia').component('profilePage', {
               UserService.getLoginUrlAsync().then(
                 function(loginUrl) {
                   if (loginUrl) {
-                    window.location.href = loginUrl;
+                    WindowRef.nativeWindow.location.href = loginUrl;
```

Comment:
Why do we need these changes?

Reply:
I changed `$window` and `window` to `WindowRef` in order to be easier to understand and test the changes made in `window`.